base/printf.jl

```diff
@@ -650,11 +656,13 @@ function gen_s(flags::String, width::Int, precision::Int, c::Char)
             push!(blk.args, pad(width, :($width-strwidth($x)), ' '))
         end
     else
+        push!(blk.args, :(io = IOBuffer()))
```

Comment:
This looks like it could have significant overhead --- maybe use the old version of the code when precision is known to be -1?